Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift

```diff
@@ -10,7 +10,7 @@ import Deferred
 
 private let MAP_SPAN_DELTA = 0.05
 private let MAP_LATITUDE_OFFSET = 0.015
-
+private let LOADING_TIMEOUT_IN_SECONDS: TimeInterval = 10
```

Comment:
Place Loading times out when `placesProviderDidFinishFetchingPlaces` is called, and the amount of time it continues to try and fetch places for is specified in `PlacesProvider.numberOfRetries` (which really should be moved to remote config). It is currently set to 60, which means the place fetching will try every second for `numberOfRetries` for as long as we are receiving either no places or the number of places we are fetching continues to rise. This is so that if it is taking the server a long time to gather the data for an area and populate the database we continue to try while we fetch.

With this timeout implementation we could tell people we found no places after 10 seconds and then start receiving data after 11 seconds. I'm not sure how that would feel.

It makes me wonder whether the place fetching timeout should be moved to `PlacesProvider`. If we have retried 10 times without receiving any places, stop fetching places and call `placesProviderDidFinishFetchingPlaces`. If we have started to receive places we can continue to loop in our fetching until we have received them all.

Does that make sense?